src/main/java/com/amazonaws/services/kinesis/connectors/s3/S3ZipEmitter.java

```diff
@@ -0,0 +1,75 @@
+package com.amazonaws.services.kinesis.connectors.s3;
+
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import com.amazonaws.services.kinesis.connectors.KinesisConnectorConfiguration;
+import com.amazonaws.services.kinesis.connectors.UnmodifiableBuffer;
+import com.amazonaws.services.kinesis.connectors.s3.S3Emitter;
+
+public class S3ZipEmitter extends S3Emitter {
+    private static final Log LOG = LogFactory.getLog(S3ZipEmitter.class);
+
+    public S3ZipEmitter(KinesisConnectorConfiguration configuration) {
+        super(configuration);
+    }
+
+    @Override
+    protected String getS3FileName(String firstSeq, String lastSeq) {
+        return firstSeq + "-" + lastSeq + ".zip";
+    }
+
+    @Override
+    public List<byte[]> emit(final UnmodifiableBuffer<byte[]> buffer) throws IOException {
+
+        ArrayList<byte[]> compressedRecords = new ArrayList<byte[]>();
+
+        List<byte[]> records = buffer.getRecords();
+
+        // Write all of the records to a compressed output stream
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        for (byte[] record : records) {
+            try {
+                baos.write(record);
+            } catch (Exception e) {
+                LOG.error("Error writing record to output stream. Failing this emit attempt. Record: "
+                                + Arrays.toString(record),
+                        e);
+                return buffer.getRecords();
+            }
+        }
+
+        // Get the Amazon S3 filename
+        String s3FileName = super.getS3FileName(buffer.getFirstSequenceNumber(), buffer.getLastSequenceNumber());
+
+        compressedRecords.add(this.compress(s3FileName, baos.toByteArray()));
```

Comment:
Sorry, I misunderstood your earlier question. The redshiftbasic emitter will not work with this. It needs to be modified to uncompress before reading the file. 
My main intention was not specifically for use with redshift connector. Rather a basic s3 connector where raw data can be archived. If manifest functionlity is added (straight forward), other lazy consumers can read from s3 and do further processing.